pkg/domain/infra/abi/play.go

```diff
@@ -964,3 +978,55 @@ func (ic *ContainerEngine) PlayKubeDown(ctx context.Context, body io.Reader, _ e
 
 	return reports, nil
 }
+
+// playKubeSecret allows users to create and store a kubernetes secret as a podman secret
+func (ic *ContainerEngine) playKubeSecret(secret *v1.Secret) (*entities.SecretCreateReport, error) {
+	r := &entities.SecretCreateReport{}
+
+	// Create the secret manager before hand
+	secretsManager, err := ic.Libpod.SecretsManager()
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := yaml.Marshal(secret)
+	if err != nil {
+		return nil, err
+	}
+
+	secretsPath := ic.Libpod.GetSecretsStorageDir()
+	opts := make(map[string]string)
+	opts["path"] = filepath.Join(secretsPath, "filedriver")
+	// maybe k8sName(data)...
+	// using this does not allow the user to use the name given to the secret
+	// but keeping secret.Name as the ID can lead to a collision.
+
+	s, err := secretsManager.Lookup(secret.Name)
+	if err == nil {
+		if val, ok := s.Metadata["immutable"]; ok {
+			if val == "true" {
+				return nil, fmt.Errorf("cannot remove colliding secret as it is set to immutable")
```

Comment:
```suggestion
				return nil, fmt.Errorf("cannot modify existing secret as it is set to immutable")
```
Shouldn't this be modify ? I think at this point user is expecting to secret get created or get modified.

Reply:
I think `remove colliding` makes more sense since that's what is happening here on a podman level. We remove the colliding secret if we can ahd recreate a new one with possibly completely different config.

Reply:
@cdoern Actually afaik k8s gives a error msg something on the lines of "unable to modify immutable field" but this is not worth blocking the PR. Could you leave this comment unresolved so I can confirm how it looks on k8s and  will update here and open a PR if need to.